test/mwcas_descriptors_test.cpp

```diff
@@ -127,7 +129,7 @@ class MwCASDescriptorFixture : public ::testing::Test
   MwCAS(  //
       const MwCASTargets &targets)
   {
-    if constexpr (std::is_same_v<MwCASDesc, DLFMwCAS>) {
+    if constexpr (std::is_same_v<MwCASDesc, DLFMwCAS> || std::is_same_v<MwCASDesc, LFMwCAS>) {
```

Comment:
ロックフリーMwCASは記述子を他のスレッドと共有する必要がある（i.e., 記述子をヒープ領域に用意する必要がある）ので，ここの変更は不要です．このブロックの書き方だと局所変数として各スレッドのスタック領域に記述子が用意されるため，MwCAS終了後に即座に領域がリリースされて，ダングリングポインタが発生します．

逆に言うと，他のロックフリー実装と同様に `GetDescriptor` 関数を用意してあげる必要があります．一方で，GCは参照カウンタベースで行うので，エポック用のガード生成は不要です．つまり，完全に専用の分岐を用意してあげる必要があります．

```suggestion
    if constexpr (std::is_same_v<MwCASDesc, LFMwCAS>) {
      while (true) {
        auto *desc = MwCASDesc::GetDescriptor();
        for (auto idx : targets) {
          auto *addr = &(target_fields_[idx]);
          const auto cur_val = MwCASDesc::template Read<Target>(addr, kRelaxed);
          const auto new_val = cur_val + 1;
          desc->AddMwCASTarget(addr, cur_val, new_val, kRelaxed);
        }
        if (desc->MwCAS()) return;
      }
    } else if constexpr (std::is_same_v<MwCASDesc, DLFMwCAS>) {
```